lib/HeapArrayITE.ml

```diff
@@ -103,13 +105,15 @@ module M : Heap_intf.M with type vt = Term.t = struct
         | Val (Loc l) -> (
             match Hashtbl.find_opt tbl l with
             | Some arr ->
+                (* Printf.printf "Index: %s\n" (Term.to_string index);
+                Printf.printf "Array: %s\n" (String.concat ", " (Array.to_list (Array.map Term.to_string arr))); *)
                 let aux =
                   Array.of_list
                     (List.filteri
                        (fun index' _ ->
                          (* can be optimized *)
                          let e = Binop (Equals, index, Val (Integer index')) in
-                         if Translator.is_sat ([ e ] @ pc) then true else false)
+                         Translator.is_sat ([ e ] @ pc))
```

Comment:
```suggestion
                         Translator.is_sat (e :: pc))
```
I know you didn't change this, but here we can use list cons instead of concat